base/logging_new_config.go

```diff
@@ -0,0 +1,218 @@
+package base
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/natefinch/lumberjack"
+	"github.com/pkg/errors"
+)
+
+const (
+	errorMinAge = 180
+	warnMinAge  = 90
+	infoMinAge  = 3
+	debugMinAge = 1
+)
+
+type LoggingConfig struct {
+	LogFilePath    string         `json:",omitempty"` // Absolute or relative path on the filesystem to the log file directory. A relative path is from the directory that contains the Sync Gateway executable file.
+	RedactionLevel RedactionLevel `json:",omitempty"`
+
+	Console LogConsoleConfig `json:",omitempty"` // Console Output
+
+	Error LogFileConfig `json:",omitempty"` // Error log file output
+	Warn  LogFileConfig `json:",omitempty"` // Warn log file output
+	Info  LogFileConfig `json:",omitempty"` // Info log file output
+	Debug LogFileConfig `json:",omitempty"` // Debug log file output
+
+	DeprecatedDefaultLog *LogAppenderConfig `json:"default,omitempty"` // Deprecated "default" logging option.
+}
+
+type LogConsoleConfig struct {
+	LogLevel     *LogLevel `json:",omitempty"` // Log Level for the console output
+	LogKeys      []string  `json:",omitempty"` // Log Keys for the console output
+	ColorEnabled bool      `json:",omitempty"` // Log with color for the console output
+
+	logKey *LogKey
+	output io.Writer
+	logger *log.Logger
```

Comment:
Our usual pattern for Config structs is that they are the straight representation of the JSON config.  Defining things like logger, output (and maybe LogKey) in the config feels unexpected.  What's the motivation for storing these here?

Reply:
As discussed, I'll refactor to embed the config inside an instance struct instead.